### PR TITLE
Fix broken error output logic

### PR DESCRIPTION
**Pull request by victor-onofrei**

### CODE REVIEW

File: Resolve-PendingMigrations.ps1 [hunk 1]
```diff
@@ -29,9 +29,41 @@ foreach ($mailbox in $allMailboxes) {
             Set-RemoteMailbox $mailboxInfo.UserPrincipalName -EmailAddresses $mailboxInfo.EmailAddresses `
              -EmailAddressPolicyEnabled $false
         } else {
-            Write-Warning "Mailbox $mailbox has on-premise archive"
+            [System.String]$message = "Mailbox $mailbox has on-premise archive enabled. Processing the script requires permanently " +
+            "disabling the archive which will result in data loss. Consider merging the 2 mailbox objects manually after backup. Script aborting..."
```
Comment (revolter) [inline Resolve-PendingMigrations.ps1:33]: Indent this line with 4 spaces

File: Resolve-PendingMigrations.ps1 [hunk 2]
```diff
@@ -29,9 +29,41 @@ foreach ($mailbox in $allMailboxes) {
             Set-RemoteMailbox $mailboxInfo.UserPrincipalName -EmailAddresses $mailboxInfo.EmailAddresses `
              -EmailAddressPolicyEnabled $false
         } else {
-            Write-Warning "Mailbox $mailbox has on-premise archive"
+            [System.String]$message = "Mailbox $mailbox has on-premise archive enabled. Processing the script requires permanently " +
+            "disabling the archive which will result in data loss. Consider merging the 2 mailbox objects manually after backup. Script aborting..."
+            [System.Management.Automation.PSInvalidCastException]$exception = New-Object -TypeName System.Management.Automation.PSInvalidCastException `
+                -ArgumentList `
+                    $message
```
Comment (revolter) [inline Resolve-PendingMigrations.ps1:36]:
```suggestion
            [System.Management.Automation.PSInvalidCastException]$exception = New-Object `
                -TypeName System.Management.Automation.PSInvalidCastException `
                -ArgumentList $message
```

File: Resolve-PendingMigrations.ps1 [hunk 3]
```diff
@@ -29,9 +29,41 @@ foreach ($mailbox in $allMailboxes) {
...
+            "disabling the archive which will result in data loss. Consider merging the 2 mailbox objects manually after backup. Script aborting..."
+            [System.Management.Automation.PSInvalidCastException]$exception = New-Object -TypeName System.Management.Automation.PSInvalidCastException `
+                -ArgumentList `
+                    $message
+            [System.Management.Automation.ErrorRecord]$errorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord `
+                -ArgumentList `
+                    $exception,
+                    'ArchiveEnabled',
+                    ([System.Management.Automation.ErrorCategory]::PermissionDenied),
+                    $mailbox
```
Comment (revolter) [inline Resolve-PendingMigrations.ps1:42]:
```suggestion
            [System.Management.Automation.ErrorRecord]$errorRecord = New-Object `
                -TypeName System.Management.Automation.ErrorRecord `
                -ArgumentList `
                    $exception,
                    'ArchiveEnabled',
                    ([System.Management.Automation.ErrorCategory]::PermissionDenied),
                    $mailbox
```

File: Resolve-PendingMigrations.ps1 [hunk 4]
```diff
@@ -29,9 +29,41 @@ foreach ($mailbox in $allMailboxes) {
...
+            $errorArguments = @{
+                Exception = $errorRecord.Exception
+                ErrorId = $errorRecord.FullyQualifiedErrorId
+                TargetObject = $errorRecord.TargetObject
+            }
+            Write-Error @errorArguments
         }
     } else {
-        Write-Warning "Mailbox $mailbox has on-premise content"
+        [System.String]$message = "Mailbox $mailbox has on-premise content above threshold. Processing the script requires permanently " +
```
Comment (revolter) [inline Resolve-PendingMigrations.ps1:51]: This duplicates an existing variable. Merge and move them to line `22`.

File: Resolve-PendingMigrations.ps1 [hunk 5]
```diff
@@ -29,9 +29,41 @@ foreach ($mailbox in $allMailboxes) {
...
+        "disabling the mailbox which will result in data loss. Consider merging the 2 mailbox objects manually after backup. Script aborting..."
+        [System.Management.Automation.PSInvalidCastException]$exception = New-Object -TypeName System.Management.Automation.PSInvalidCastException `
+            -ArgumentList `
+                $message
```
Comment (revolter) [inline Resolve-PendingMigrations.ps1:55]: Same for this one.

File: Resolve-PendingMigrations.ps1 [hunk 6]
```diff
@@ -29,9 +29,41 @@ foreach ($mailbox in $allMailboxes) {
...
+        [System.Management.Automation.ErrorRecord]$errorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord `
+            -ArgumentList `
+                $exception,
+                'ContentAboveThreshold',
+                ([System.Management.Automation.ErrorCategory]::PermissionDenied),
+                $mailbox
```
Comment (revolter) [inline Resolve-PendingMigrations.ps1:61]:
```suggestion
        [System.Management.Automation.ErrorRecord]$errorRecord = New-Object `
            -TypeName System.Management.Automation.ErrorRecord `
            -ArgumentList `
                $exception,
                'ContentAboveThreshold',
                ([System.Management.Automation.ErrorCategory]::PermissionDenied),
                $mailbox
```

File: Resolve-PendingMigrations.ps1 [hunk 7]
```diff
@@ -29,9 +29,41 @@ foreach ($mailbox in $allMailboxes) {
...
+            $errorArguments = @{
+                Exception = $errorRecord.Exception
+                ErrorId = $errorRecord.FullyQualifiedErrorId
+                TargetObject = $errorRecord.TargetObject
+            }
+            Write-Error @errorArguments
```
Comment (revolter) [inline Resolve-PendingMigrations.ps1:67]: Unindent these lines.

**Review by revolter — CHANGES_REQUESTED** [review]

File: Resolve-PendingMigrations.ps1 (hunk 1 above)
Comment (revolter) [inline Resolve-PendingMigrations.ps1:33]: "Skipping this mailbox." instead of "Script aborting..."

File: Resolve-PendingMigrations.ps1 (hunk 4 above)
Comment (victor-onofrei) [reply Resolve-PendingMigrations.ps1:52]: Not duplicate, different message.

File: Resolve-PendingMigrations.ps1 (hunk 5 above)
Comment (victor-onofrei) [reply Resolve-PendingMigrations.ps1:56]: Not duplicate, different message.

File: Resolve-PendingMigrations.ps1 (hunk 7 above)
Comment (victor-onofrei) [reply Resolve-PendingMigrations.ps1:69]: Removed superfluous array.

File: Resolve-PendingMigrations.ps1 (hunk 4 above)
Comment (revolter) [reply Resolve-PendingMigrations.ps1:47]: Oh, ok.

File: Resolve-PendingMigrations.ps1 (hunk 5 above)
Comment (revolter) [reply Resolve-PendingMigrations.ps1:51]: Right

File: Resolve-PendingMigrations.ps1 [hunk 8]
```diff
@@ -29,9 +29,38 @@ foreach ($mailbox in $allMailboxes) {
...
+                    $mailbox
+            Write-Error -ErrorRecord $errorArguments
         }
     } else {
-        Write-Warning "Mailbox $mailbox has on-premise content"
+        [System.String]$message = "Mailbox $mailbox has on-premise content above threshold. Processing the script requires permanently " +
+        "disabling the mailbox which will result in data loss. Consider merging the 2 mailbox objects manually after backup. Script aborting..."
+        [System.Management.Automation.PSInvalidCastException]$exception = New-Object -TypeName System.Management.Automation.PSInvalidCastException `
+            -ArgumentList `
+                $message
```
Comment (revolter) [inline Resolve-PendingMigrations.ps1:51]:
```suggestion
        [System.Management.Automation.PSInvalidCastException]$exception = New-Object `
            -TypeName System.Management.Automation.PSInvalidCastException `
            -ArgumentList $message
```

File: Resolve-PendingMigrations.ps1 (hunk 7 above)
Comment (revolter) [reply Resolve-PendingMigrations.ps1]: Unindent the line.

**Pull request merged**